From f8fa2196370cc21e78f2a3e1a38582ba1a7269b1 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Thu, 21 Nov 2013 10:31:06 +0800
Subject: [PATCH 17/45] ENGR00288965 after recored many times, can't recorder
 video in the clip

In JNICameraContext::postData, should release record frame

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 core/jni/android_hardware_Camera.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/jni/android_hardware_Camera.cpp b/core/jni/android_hardware_Camera.cpp
index 4f44c26..97aea07 100644
--- a/core/jni/android_hardware_Camera.cpp
+++ b/core/jni/android_hardware_Camera.cpp
@@ -15,6 +15,10 @@
 ** limitations under the License.
 */
 
+/*
+ * Copyright (C) 2013 Freescale Semiconductor, Inc.
+ */
+
 //#define LOG_NDEBUG 0
 #define LOG_TAG "Camera-JNI"
 #include <utils/Log.h>
@@ -313,6 +317,8 @@ void JNICameraContext::postData(int32_t msgType, const sp<IMemory>& dataPtr,
     switch (dataMsgType) {
         case CAMERA_MSG_VIDEO_FRAME:
             // should never happen
+            // actually, it happeded little change on start/stop recorder
+            mCamera->releaseRecordingFrame(dataPtr);
             break;
 
         // For backward-compatibility purpose, if there is no callback
-- 
1.8.0

